### PR TITLE
[PM-5364] Create SSO Login Service and add state ownership

### PR DESCRIPTION
**Pull request by jlf0dev**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Creates a SSO Login Service for holding state during login for sso related purposes. This service doesn't contain any logic yet as that would expand testing, and only tries to make a minimal amount of changes.

There are no migrations for the new state as they are only used as a cache during a sso login flow
## Code changes

- **file.ext:** Description of what was changed and why

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Review by justindbaur — DISMISSED** [review]
There are no migrations for this disk state because this data is expected to be cleared after a successful login and it's incredibly unlikely/not worth the migration that someone updates the extension halfway through the SSO process right?

If that is the case this gets approval from platform. 

Very nice documentation!

**Review by JaredSnider-Bitwarden — COMMENTED** [review]
Excellent work on this! Just a few questions below 👇

File: libs/angular/src/auth/components/base-login-decryption-options.component.ts [hunk 1]
```diff
@@ -161,7 +163,7 @@ export class BaseLoginDecryptionOptionsComponent implements OnInit, OnDestroy {
 
   async loadNewUserData() {
```
[the remark this answers is not in the document]
Comment (jlf0dev) [reply libs/angular/src/auth/components/base-login-decryption-options.component.ts:164]: Yes, ran through the TDE user autoprovision flow to make sure the decryption options works correctly.

File: libs/common/src/auth/abstractions/sso-login.service.abstraction.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+   */
+  setSsoState: (ssoState: string) => Promise<void>;
+  /**
+   * Gets the value of the user's organization identifier.
+   *
+   * This should only be used during the SSO flow to identify the organization that the user is attempting to log in to.
+   * Do not use this value outside of the SSO login flow.
+   * @returns The user's organization identifier.
+   */
+  getOrganizationIdentifier: () => Promise<string>;
```
[the remark this answers is not in the document]
Comment (jlf0dev) [reply libs/common/src/auth/abstractions/sso-login.service.abstraction.ts]: Good call, updated

**Review by JaredSnider-Bitwarden — APPROVED** [review]
LGTM!

**Review by JaredSnider-Bitwarden — APPROVED** [review]
Nice work! Tests passing!

**Review by justindbaur — DISMISSED** [review]

**Review by shane-melton — DISMISSED** [review]
Changes owned by Vault look good! 

With that said, should Vault even own the `link-sso.directive`?

**Comment by jlf0dev** [comment]
@shane-melton @justindbaur @JaredSnider-Bitwarden Had to resolve some conflicts, just need a review for merging. Thanks!

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by shane-melton — APPROVED** [review]

**Review by justindbaur — CHANGES_REQUESTED** [review]

File: libs/common/src/platform/services/state.service.ts [hunk 3]
```diff
@@ -2446,77 +2446,6 @@ export class StateService<
...
-  async setSsoOrganizationIdentifier(value: string, options?: StorageOptions): Promise<void> {
-    const globals = await this.getGlobals(
-      this.reconcileOptions(options, await this.defaultOnDiskLocalOptions()),
-    );
-    globals.ssoOrganizationIdentifier = value;
-    await this.saveGlobals(
-      globals,
-      this.reconcileOptions(options, await this.defaultOnDiskLocalOptions()),
-    );
-  }
```
[the remark this answers is not in the document]
Comment (jlf0dev) [reply libs/common/src/platform/services/state.service.ts:2481]: Our sso flow currently takes you back to bitwarden in the same tab, so I can't think of any reason to use local storage, can you? This has also been through QA as well.

[the remark this answers is not in the document]
Comment (justindbaur) [reply libs/common/src/platform/services/state.service.ts:2481]: That's fine with me then, just making sure.

**Review by justindbaur — APPROVED** [review]

**Pull request merged**